mathics_scanner/tokeniser.py

```diff
@@ -148,7 +148,7 @@
     ("ConjugateTranspose", r" \uf3c9 "),
     ("HermitianConjugate", r" \uf3ce "),
     ("Integral", r" \u222b "),
-    ("DifferentialD", r" \U0001D451 "),
+    ("DifferentialD", r" \u1d451 | \uf74c"),
```

Comment:
This is something about I am not completely sure. I guess that `\[DifferentialD]` should be parsed as `\uf74c` for the internal representation, and then it makes sense that a match with `\uf74c` should represent a match with the operator. `\u1d451`, on the other hand, is a printable representation for  `\[DifferentialD]`, so it does not need to match the operator in general parsing.

Reply:
If I recall correctly the intent was to allow for different kinds of input devices sending different kinds of codes. 

In this situation and the one above, it may be that you are being myopic - you see a problem and consider only the aspect that you are running into problems with while possibly not understanding the larger picture. 

With respect to the table change things to think about are not just the existing kind of tagging, but changes can be made in the translator to specific tables. So there might be a table that is custom for mathics-core, but mathics-scanner gets something similar but different. 

And we can also consider expanding the fields in the YML file as has been done a number of times in the past as we discover different kinds of needs. For example operator information was recently added, before that unicode equivalences for non-standard WL symbols. 

At any rate I need some time to look this over and the history and think about this and that may happen on the weekend.

Reply:
> If I recall correctly the intent was to allow for different kinds of input devices sending different kinds of codes.

Ok, but whatever is the input, what we want to pass to the tokenizer is a uniform wl string. Then, after all the evaluations are carried on, in the output, we want to translate l the wl string to an encoding that the interpreter knows how to handle. 


> 
> In this situation and the one above, it may be that you are being myopic - you see a problem and consider only the aspect that you are running into problems with while possibly not understanding the larger picture.
> 
> With respect to the table change things to think about are not just the existing kind of tagging, but changes can be made in the translator to specific tables. So there might be a table that is custom for mathics-core, but mathics-scanner gets something similar but different.
> 
> And we can also consider expanding the fields in the YML file as has been done a number of times in the past as we discover different kinds of needs. For example operator information was recently added, before that unicode equivalences for non-standard WL symbols.
> 
> At any rate I need some time to look this over and the history and think about this and that may happen on the weekend.

Reply:
> Ok, but whatever is the input, what we want to pass to the tokenizer is a uniform wl string. 

Yes, that is a _new_ and _different_ situation.  A couple possibilities come to mind that don't change this particular piece of code, but I'd have to think about it some more.

Reply:
Given what I wrote in https://github.com/Mathics3/mathics-scanner/pull/43#discussion_r962317757 this change is okay as is.

Reply:
Also, I'd note in a comment that `\u1d451` is to handle standard Unicode while ` \uf74c` is there to handle WL-specific unicode.